### PR TITLE
[MTSRE-515] feat: introduce monitoringStack API to the metadata

### PR DESCRIPTION
**Pull request by yashvardhan-kukreja**
Signed-off-by: Yashvardhan Kukreja <ykukreja@redhat.com>

# py-mtcli

## Description

introduces the support for the `monitoringStack` field in the metadata schema for specifying the config for the monitoring stack CR to be utilised by the addon-operator in runtime. 

## Checklist

**For any modification to `managedtenants/bundles/`**

- [ ] `$ managedtenants --addons-dir=../managed-tenants-bundles/addons --dry-run bundles` (successfuly built all addons)
- [ ] `$ managedtenants -addons-dir=../managed-tenants-bundles/addons --addon-name=<addon> bundles --quay-org <personal_org>` (successfuly built and push a single addon)

### CODE REVIEW

File: managedtenants/schemas/metadata.schema.yaml [hunk 1]
```diff
@@ -129,6 +129,39 @@ properties:
...
+        description: "List of series names to federate from the prometheus server."
+      matchLabels:
+        type: object
+        items:
+          type: string
+          format: printable
+          pattern: ^[a-zA-Z_][a-zA-Z0-9_]*$
+        description: "List of labels used to discover the prometheus server(s) to be federated."
+        minItems: 1
+  metricsFederation:
```
Comment (ashishmax31) [inline managedtenants/schemas/metadata.schema.yaml:164]: Why is this renamed to `metricsFederation`?

File: tests/sss/test_federated_metrics.py [hunk 2]
```diff
@@ -8,6 +8,94 @@
 from managedtenants.core.addon_manager import AddonManager
 from managedtenants.core.addons_loader.sss import Sss
 
+@given(data=hypothesis_strategies.data())
+def test_metrics_federation_v1(data):
```
Comment (ashishmax31) [inline tests/sss/test_federated_metrics.py:12]: This was existing functionality right? You just added tests for them right?

File: managedtenants/schemas/metadata.schema.yaml (hunk 1 above)
Comment (yashvardhan-kukreja) [reply managedtenants/schemas/metadata.schema.yaml:164]: previously metrics-federation config was provided under the `monitoring` section. And after realising that we need a new `monitoringStack` field, it was realised that `monitoring` and `monitoringStack` fields both in the addon metadata is pretty bad naming. Hence, the idea of renaming/deprecating the `monitoring` field to `metricsFederation` was formed.

That's why, if you notice, the description of the `monitoring` field begins with `[DEPRECATED]`

PS: the `monitoringStack` is totally unrelated to these `monitoring` and `metricsFederation` fields by the way

File: tests/sss/test_federated_metrics.py (hunk 2 above)
Comment (yashvardhan-kukreja) [reply tests/sss/test_federated_metrics.py:12]: These tests were added to ensure that the selectorsyncsets are rendered properly. Just added them for the sake of better coverage in favor of backwards compatibility. Although it's pretty much not possible for us to move back to SelectorSyncSet, no harm in having some extra tests ;)

**Review by ashishmax31 — APPROVED** [review]
